### PR TITLE
SLE-252 Add performance metrics to telemetry

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java [hunk 1]
```diff
@@ -185,12 +186,29 @@ private void runAnalysisAndUpdateMarkers(@Nullable Server server, Map<ISonarLint
     Map<ISonarLintIssuable, List<Issue>> issuesPerResource = new LinkedHashMap<>();
     filesToAnalyze.keySet().forEach(slFile -> issuesPerResource.put(slFile, new ArrayList<>()));
 
+    long start = System.currentTimeMillis();
     AnalysisResults result = run(server, config, issuesPerResource, monitor);
     if (!monitor.isCanceled() && result != null) {
+      updateTelemetry(inputFiles, start);
```
Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java]: Is it intended to not count marker processing in the analysis time? IMO we should count everything

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java [hunk 2]
```diff
@@ -185,12 +186,29 @@ private void runAnalysisAndUpdateMarkers(@Nullable Server server, Map<ISonarLint
...
+  private void updateTelemetry(List<ClientInputFile> inputFiles, long start) {
+    SonarLintTelemetry telemetry = SonarLintCorePlugin.getTelemetry();
+    if (inputFiles.size() == 1) {
+      telemetry.analysisDoneOnSingleFile(getExtension(inputFiles.iterator().next()), (int) (System.currentTimeMillis() - start));
+    } else {
+      telemetry.analysisDoneOnMultipleFiles();
+    }
+  }
+
+  private String getExtension(ClientInputFile next) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java]: Code Smell Code Smell: Make "getExtension" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.eclipse%3Asonarlint-eclipse-parent&pullRequest=142&issues=AWKWOi7h69PS8Rxb2jtJ&open=AWKWOi7h69PS8Rxb2jtJ)

[ISSUE_KEY]: AWKWOi7h69PS8Rxb2jtJ

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java (hunk 1 above)
Comment (janos-ss) [reply org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java]: Yes that's what I intended. I go with your opinion, I moved down the line.

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java [hunk 3]
```diff
@@ -185,12 +186,29 @@ private void runAnalysisAndUpdateMarkers(@Nullable Server server, Map<ISonarLint
...
 
+    long start = System.currentTimeMillis();
     AnalysisResults result = run(server, config, issuesPerResource, monitor);
     if (!monitor.isCanceled() && result != null) {
       updateMarkers(server, docPerFiles, issuesPerResource, result, triggerType, monitor);
+      updateTelemetry(inputFiles, start);
     }
   }
 
+  private void updateTelemetry(List<ClientInputFile> inputFiles, long start) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java]: Code Smell Code Smell: Make "updateTelemetry" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.eclipse%3Asonarlint-eclipse-parent&pullRequest=142&issues=AWKWRRav-RUO4oGVXYul&open=AWKWRRav-RUO4oGVXYul)

[ISSUE_KEY]: AWKWRRav-RUO4oGVXYul

**Pull request merged**